### ISSUE TITLE
Boost dependency(?)

### ISSUE THREAD

**Issue by uvilla** [conda-forge/valgrind-feedstock#7]
Why does `valgrind-feedstock` depend on boost?

I looked on the `valgrind` manual and there's no reference to boost.

I noted this because calling `conda install valgrind` wanted to downgrade my version of boost (and causing a chain reaction of changes).

**Comment by notestaff** [comment]
configure.ac does check for Boost.

You can install valgrind in its own separate env, and make a shim in your main env to activate the valgrind env and run valgrind, like so:

conda create -n valgrind_env valgrind -y

then, put a file like this in your working env's bin/ directory:

#!/bin/bash

set -e -o pipefail

. "/ilya/miniconda3/etc/profile.d/conda.sh"
conda activate valgrind_env
valgrind "$@"
conda deactivate

**Comment by uvilla** [comment]
Thanks for your explanation.

I see what you are referring too in `configure.ac`:

```
#----------------------------------------------------------------------------
# Other library checks
#----------------------------------------------------------------------------
# There now follow some tests for Boost, and OpenMP.  These
# tests are present because Drd has some regression tests that use
# these packages.
```

From my understanding, `valgrind` doesn't really need `boost` besides running some tests. 
So should not `boost-cpp` be a `build` requirement and not `host` or `run` requirement?

**Comment by jakirkham** [comment]
How did you come to this understanding?

**Comment by uvilla** [comment]
The comments I quote explicitly says that boost is only needed for regression testing of the *DRD: a thread error detector* component of valgrind.

A recursive grep in the valgrind source will indeed show that the only place where boost is used is in the test`drd/tests/boost_thread.cpp`.

**Comment by notestaff** [comment]
@uvilla I think you're right; I'm testing a boost-less PR right now - https://github.com/conda-forge/valgrind-feedstock/pull/8
thanks,
ilya

**Comment by jakirkham** [comment]
> The comment I quoted explicitly says that boost is only needed for regression testing of the _DRD: a thread error detector_ component of valgrind.

It says there are tests for Boost. Not that it is not needed for anything else. Hence the question. 😉

> A recursive grep in the valgrind source will indeed show that the only place where boost is used is in the test`drd/tests/boost_thread.cpp`.

That said, this does answer it. Thanks for checking. 🙂

**Comment by notestaff** [comment]
@uvilla I've merged the PR that drops the Boost dependency.  (It might take a day to show up in conda search).  I had to temporarily drop the MacOS build, until I figure out an issue with it; hope you're working on Linux.

Oddly, http://www.linuxfromscratch.org/blfs/view/cvs/general/valgrind.html thinks Boost is needed not just for tests; I'll check with them for their reason.

Btw, for a low-level tool like valgrind, you _might_ get better performance if you build it from source specifically for your system, rather than using conda-forge's built-for-lowest-common-denominator version.

Thanks again for pointing out the spurious Boost dependency!
